app/src/main/java/com/example/superimagesviewer/FacebookLoginFragment.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.superimagesviewer
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.facebook.CallbackManager
+import com.facebook.login.LoginManager
+
+class FacebookLoginFragment : Fragment() {
+
+    private lateinit var callbackManager: CallbackManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        callbackManager = CallbackManager.Factory.create()
```

Comment:
Not sure if it's OK with Facebook SDK, but such kind of business logic should be moved to ViewModel

Reply:
done